### DISCUSSION TITLE
Problem with SSH key

### DISCUSSION THREAD

**Discussion by IOBerger** [community/community#22207] [New to GitHub]
<p>Hi! I am trying to enter SSH key on GitHub. I generated it with “ssh-keygen” command. And I got a message “Key is invalid. You must supply a key in OpenSSH public key format”. What should I do? My SSH key is a big amount of symbols, as it should be. It doesn’t start with “ssh-rsa” or other words, described in GitHub. How should I get the key in correct form?</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="IOBerger" data-post="1" data-topic="135230">
<div class="title">
<div class="quote-controls"></div>
 IOBerger:</div>
<blockquote>
<p>I generated it with “ssh-keygen” command.</p>
</blockquote>
</aside>
<p>That command is the right one, but could you add the complete command?</p>
<aside class="quote no-group" data-username="IOBerger" data-post="1" data-topic="135230">
<div class="title">
<div class="quote-controls"></div>
 IOBerger:</div>
<blockquote>
<p>My SSH key is a big amount of symbols, as it should be. It doesn’t start with “ssh-rsa” or other words, described in GitHub.</p>
</blockquote>
</aside>
<p>Then it’s likely you’re not copying the right file. The filename should end with <code>.pub</code>. <code>ssh-keygen</code> should have asked you where to store the file, if you accepted the default it should be named something like <code>id_rsa.pub</code> (if using RSA).</p>

**Comment by kaiser2** [comment]
<p>I am having the same issue and Yes i am copying the .pub file and not the pvt key . Still it won’t work . I tried everything , generated and regenerated the key multiple times. Any help plz. I am on windows</p>

**Reply by AnxiousLeopard** [reply]
I just had the same issue and realised that there is a Microsoft Publisher document appearing when the file is marked as ".pub"
If you right click this and open it in notepad you will have your public key.
